### DISCUSSION TITLE
Ability to cache props in getStaticProps while in development

### DISCUSSION THREAD

**Discussion by dcporter44** [vercel/next.js#21194] [Ideas]
**Describe the feature you'd like to request**

In development, `getStaticProps` runs every page load and every refresh. This is usually fine, but it becomes very annoying in development when making requests to an external API (or any other slow process) in `getStaticProps`. It would be great if we could have the ability to cache selected props during dev build (`npm run dev`). That way the development server speed is more accurate to the production build. And simply to speed up development.

If the data becomes stale during development, I'm sure developers would prefer to restart the development server than have to wait to re-fetch the data every page change like it does now. Personally, I'm fetching my i18n strings for each page from an external API (as many others do who use Translation Management Systems), and I'd prefer to only re-fetch the strings when I need to.

**Describe the solution you'd like**

I think this would best be done as a property on the returned object in `getStaticProps`. Just a simple array of prop names that should be cached during dev build. It would be great to have something like this:

```
export async function getStaticProps() {
  const res = await fetch('https://.../posts')
  const posts = await res.json()

  return {
    props: {
      posts,
    },
    devCache: ['posts']
  }
}
```

**Describe alternatives you've considered**

I've tried caching props myself by having a global `cachedProps` object that stores props I receive from external API's. Then I have a function in `getStaticProps` that uses the cache if the cache exists.  But this still re-fetches everytime the browser refreshes. It's only useful in the same session when visiting pages you've already visited. There's no way currently to do this at build time in development.

**Comment by edjw** [comment]
This would be really good.

Eleventy has something like this: https://www.11ty.dev/docs/plugins/cache

**Comment by baba43** [comment]
I'm also looking for this. 

At least it would be cool to disable the automatic cache busting, because even specifying `revalidate: false` won't stop Next.js from refetching the data over and over again.

**Comment by bebejane** [comment]
+1 for this. It's difficult to get a good developer experience when using page transitions with Framer Motion since it needs to get the data from getStaticProps each time. You can't get a feel for it until in production when content is cached. It would be great if you could easily toggle caching via env variable like: DEV_CACHE=true yarn dev

**Comment by raphtlw** [comment]
Would love to see this. It kinda gets annoying when data is always needing to be fetched from an API which takes ages to resolve. Doing this with any paid API will drive unnecessary costs and unnecessary processing cycles.

**Comment by cwtuan** [comment]
is it possible to cache the data in development via `revalidate`?

**Comment by PatrikTheDev** [comment]
I personally disagree as being able to reload gSP in dev is relatively valuable, I think you should implement a cache in the function yourself (maybe check NODE_ENV to make sure you only cache in dev). A very simple cache should be easy to implement yourself (or you could use a 3rd party package, there are plenty available)

Ideally you wouldn’t have a slow gSP but I get it’s sometimes unavoidable.
